src/components/fields/ArrayField.js

```diff
@@ -228,33 +208,35 @@ class ArrayField extends Component {
         event.preventDefault();
         event.target.blur();
       }
-      const {items} = this.state;
-      this.asyncSetState({
-        items: items.map((item, i) => {
+      const items = this.props.formData;
```

Comment:
Nit: this can alternatively be written as `const {formData: items} = this.props;`, which I personally like a lot. But here we could probably just use `formData` as it makes sense that they're a list of items in an ArrayField.